File: StartUpExample/StartUpExampleJava/app/src/main/java/com/tinlone/startupexamplejava/initializers/MapInitializer.java
package com.tinlone.startupexamplejava.initializers;

import android.content.Context;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.startup.Initializer;

import com.tinlone.startupexamplejava.dependencies.TXMap;

import java.util.Arrays;
import java.util.List;

public class MapInitializer implements Initializer<TXMap> {
    @NonNull
    @Override
    public TXMap create(@NonNull Context context) {
        Log.i("loglog", "create: MapInitializer");
        return new TXMap();
    }

    @NonNull
    @Override
    public List<Class<? extends Initializer<?>>> dependencies() {
        // map依赖了log
        return Arrays.asList(LoggerInitializer.class, SomethingInitializer.class);
    }
}

File: StartUpExample/StartUpExampleKotlin/app/src/main/java/com/tinlone/startupexamplekotlin/initializers/MapInitializer.kt
package com.tinlone.startupexamplekotlin.initializers

import android.content.Context
import android.util.Log
import androidx.startup.Initializer
import com.tinlone.startupexamplekotlin.dependencies.Logger
import com.tinlone.startupexamplekotlin.dependencies.TXMap

class MapInitializer : Initializer<TXMap> {
    override fun create(context: Context): TXMap {
        Logger.log("loglog", "create: MapInitializer")
        return TXMap()
    }

    override fun dependencies(): List<Class<out Initializer<*>?>> {
        // map依赖了log
        return listOf(LoggerInitializer::class.java, SomethingInitializer::class.java)
    }
}